lib/ae_mdw/db/sync/contract.ex

```diff
@@ -186,8 +168,73 @@ defmodule AeMdw.Db.Sync.Contract do
       {local_idx, "Channel.settle", :channel_settle_tx, _aetx, tx} ->
         Channels.settle_mutations({block_index, {call_txi, local_idx}}, tx)
 
+      {_local_idx, fname, :contract_create_tx, _aetx, tx} when fname in @contract_create_fnames ->
+        contract_pk = :aect_create_tx.contract_pubkey(tx)
+
+        [
+          aexn_create_contract_mutation(contract_pk, block_index, call_txi),
+          ContractCreateCacheMutation.new(contract_pk, call_txi)
+          | SyncOrigin.origin_mutations(
+              :contract_call_tx,
+              nil,
+              contract_pk,
+              call_txi,
+              tx_hash
+            )
+        ]
+
       {_local_idx, _fname, _tx_type, _aetx, _tx} ->
         []
     end)
   end
+
+  defp create_contract_create_aetx(block_hash, contract_pk, owner_id, initial_amount) do
+    {:ok, contract} = :aec_chain.get_contract(contract_pk)
+    deposit = :aect_contracts.deposit(contract)
+    abi_version = :aect_contracts.abi_version(contract)
+    vm_version = :aect_contracts.vm_version(contract)
+    code = retrieve_code(contract)
+    {_tag, owner_pk} = :aeser_id.specialize(owner_id)
+    owner_nonce = Db.nonce_at_block(block_hash, owner_pk)
+    nonce_tries = owner_nonce..(owner_nonce - 100)
+
+    nonce =
+      Enum.find(
+        nonce_tries,
+        &(:aect_contracts.compute_contract_pubkey(owner_pk, &1) == contract_pk)
+      )
+
+    unless nonce do
+      raise "nonce not found for #{Enc.encode(:contract_pubkey, contract_pk)} owner #{Enc.encode(:account_pubkey, owner_pk)}"
+    end
+
+    {:ok, contract_create_aetx} =
+      :aect_create_tx.new(%{
+        owner_id: owner_id,
+        nonce: nonce,
+        code: code,
+        vm_version: vm_version,
+        abi_version: abi_version,
+        deposit: deposit,
+        amount: initial_amount - deposit,
+        gas: 0,
+        gas_price: 0,
+        call_data: "",
+        fee: 0
+      })
+
+    contract_create_aetx
+  end
+
+  defp retrieve_code(contract) do
```

Comment:
This could be reused: https://github.com/aeternity/ae_mdw/blob/master/lib/ae_mdw/contract.ex#L164

Reply:
Good call, didn't realize it existed, done now